XLS-47d-PriceOracles/README.md

```diff
@@ -0,0 +1,468 @@
+
+|FieldName | Required? | JSON Type |
+|:---------|:-----------|:---------------|
+| `base_asset` | :heavy_check_mark: | `string` |
+
+`base_asset` is the asset to be priced.
+
+|FieldName | Required? | JSON Type |
+|:---------|:-----------|:---------------|
+| `quote_asset` | :heavy_check_mark: | `string` |
+
+`quote_asset` is the denomination in which the prices are expressed.
+
+|FieldName | Required? | JSON Type |
+|:---------|:-----------|:---------------|
+| `oracles` | :heavy_check_mark: | `array` |
+
+`oracles` is an array of `oracle` objects to aggregate over. `oracle` object has two fields:
+
+- |FieldName | Required? | JSON Type |
+  |:---------|:-----------|:---------------|
+  | `account` | :heavy_check_mark: | `string` |
+
+    `account` is the Oracle's account.
+
+- |FieldName | Required? | JSON Type |
+  |:---------|:-----------|:---------------|
+  | `oracle_sequence` | :heavy_check_mark: | number |
+
+  `oracle_sequence` is a unique identifier of the Price Oracle for the given Account.
+
+|FieldName | Required? | JSON Type |
+|:---------|:-----------|:---------------|
+| `trim` | | `number` |
+
+`trim` is the percentage of outliers to trim. Valid trim range is 1-25. If this parameter is included then the API returns trimmed median in addition to the average and median.
+
+|FieldName | Required? | JSON Type |
+|:---------|:-----------|:---------------|
+| `time_threshold` | | `number` |
+
+The `time_threshold` is used to define a time range in seconds for filtering out older price data. It's an optional parameter and is 0 by default; i.e. there is no filtering in this case.
+
+The price data to aggregate is selected based on specific criteria. The most recent Price Oracle object is obtained for the specified oracles. The most recent `LastUpdateTime` among all objects is chosen as the upper time threshold. A Price Oracle object is included in the aggregation dataset if it satisfies the conditions of containing the specified `base_asset`/`quote_asset` pair, including the `AssetPrice` field, and its `LastUpdateTime` is within the time range of (upper threshold - time threshold) to the upper threshold. If a Price Oracle object doesn't contain the `AssetPrice` for the specified token pair, then up to three previous Price Oracle objects are examined and include the first one that fulfills the criteria.
+
+The `get_aggregate_price` fails if:
+
+- The oracles array size is either 0 or greater than 200.
+- The oracles array's object doesn't include `account` or `oracle_sequence` or those fields have invalid value.
+- `base_asset` or `quote_asset` are missing.
+- `trim` or `time_threshold` contain invalid uint value.
+- If the resulting data set is empty.
+
+###### Example of get_aggregate_price API JSON request
+
+    {
+    "method": "get_aggregate_price",
+    "params": [
+        {
+            "ledger_index": "current",
+            "base_asset": "XRP",
+            "quote_asset": "USD",
+            "flags": 7,
```

Comment:
what this ? Cannot see `flags` defined above.

Reply:
Thanks. It's a leftover from a first version, which did have the flags. Removed.